### DISCUSSION TITLE
brew autocomplete not working

### DISCUSSION THREAD

**Discussion by gabriel-lima96** [ohmyzsh/ohmyzsh#11934] [Help] [answered]
Hey, I just made a fresh install, added the brew plugin and restarted my terminal. However there is no completion for `brew`. When I type `brew i` then hit *Tab* I'm expecting to get the available commands for brew (info, install etc), instead it just lists the documents on my current folder.
Tried to fix it following [brew shell completion](https://docs.brew.sh/Shell-Completion) and added only the `FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"`. Closed the terminal, opened again. No luck. BUT, after I ran `compinit` manually it suddenly started working. Removed the FPATH rule from my .zshrc to see if a fresh install would behave the same and it is.

To sum it up: using a fresh install and adding the brew plugin there is no autocomplete except if I run compinit manually.

Is this a bug or am I missing something?

EDIT: Okay, I forgot to run the command after installing brew that put the "eval .../brew shellenv" in the .zprofile. Now the autocomplete works if I keep the FPATH changes inside my .zshrc, as said in the brew shell completion docs. However, wasn't it supposed to do that by default when the brew plugin is enabled?

**Comment by carlosala** [comment, marked as answer]
Adding
```zsh
autoload -Uz _brew
compdef _brew brew
```
should do the trick.

Another option is to move the FPATH entry addition before sourcing `oh-my-zsh.sh`.
